### PR TITLE
Fix inconsistent arch arguments expected in cudnn package

### PR DESCRIPTION
**Pull request by 0luhancheng0**
cuDNN package expect `aarch64sbsa` when spack uses `aarch64`

### CODE REVIEW

**Comment by adamjstewart** [comment]
This arch is used both in `when` statements and in the URL. You may need to update the `url_for_version` to ensure that the download still works.

**Comment by 0luhancheng0** [comment]
@adamjstewart thanks for pointing it out, the latest commit should fix this url issue

**Review by adamjstewart — DISMISSED** [review]
Don't have access to a system on which to test this, but it seems good to me.

File: var/spack/repos/builtin/packages/cudnn/package.py [hunk 1]
```diff
@@ -189,7 +189,7 @@ class Cudnn(Package):
     #     https://developer.nvidia.com/rdp/cudnn-archive
     # Note that download links don't work from command line,
     # need to use modified URLs like in url_for_version.
-
+# https://developer.nvidia.com/compute/machine-learning/cudnn/secure/8.2.2/11.4_07062021/cudnn-11.4-linux-aarch64sbsa-v8.2.2.26.tgz
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/cudnn/package.py]: Can you indent this to the same level as the other comments? Flake8 ignores URLs so no need to worry about line length.

Comment (0luhancheng0) [reply var/spack/repos/builtin/packages/cudnn/package.py]: sorry for the confusion, that is not suppose to be there

**Review by adamjstewart — APPROVED** [review]

**Review by adamjstewart — APPROVED** [review]

**Review by bvanessen — APPROVED** [review]

**Pull request merged**